flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/typeinfo/RowSerializer.scala

```diff
@@ -74,11 +79,16 @@ class RowSerializer(fieldSerializers: Array[TypeSerializer[Any]])
 
   override def serialize(value: Row, target: DataOutputView) {
     val len = fieldSerializers.length
-    var i = 0
-    while (i < len) {
-      val serializer = fieldSerializers(i)
-      serializer.serialize(value.productElement(i), target)
-      i += 1
+    (0 to len - 1).foreach {
+      index =>
+        val o: AnyRef = value.productElement(index).asInstanceOf[AnyRef]
+        if (o == null) {
+          target.writeBoolean(true)
+        } else {
+          target.writeBoolean(false)
+          val serializer = fieldSerializers(index)
+          serializer.serialize(value.productElement(index), target)
+        }
```

Comment:
Could you please change this back to the simple while loop. I know that using the fancy Scala features is tempting but the performance of a simple while loop should be better than creating a range and iterating over it using foreach.